File: library/src/main/java/cn/wandersnail/widget/listener/RejectFastViewClickListener.java
package cn.wandersnail.widget.listener;

import android.view.View;

import androidx.annotation.NonNull;

/**
 * date: 2019/8/22 18:16
 * author: zengfansheng
 */
public class RejectFastViewClickListener extends RejectFastClickListener implements View.OnClickListener {
    private RejectableViewClickCallback callback;

    public RejectFastViewClickListener(int minInterval, @NonNull RejectableViewClickCallback callback) {
        super(minInterval);
        this.callback = callback;
    }

    public RejectFastViewClickListener(@NonNull RejectableViewClickCallback callback) {
        super(1000);
        this.callback = callback;
    }

    @Override
    public void onClick(View view) {
        check(view, new Callback() {
            @Override
            public void onReject() {
                callback.onReject(view);
            }

            @Override
            public void onAccept() {
                callback.onAccept(view);
            }
        });
    }
}

File: library/src/main/java/com/snail/widget/listener/RejectFastViewClickListener.kt
package com.snail.widget.listener

import android.view.View
import androidx.annotation.CallSuper

/**
 * 防止快速点击的
 *
 * date: 2019-06-16 11:59
 * author: zengfansheng
 *
 * @param minInterval 最小间隔，单位：毫秒
 */
class RejectFastViewClickListener @JvmOverloads constructor(private val callback: RejectableViewClickCallback, minInterval: Long = 1000) :
    RejectFastClickListener(minInterval), View.OnClickListener {

    @CallSuper
    override fun onClick(v: View?) {
        check(v) {
            if (it) {
                callback.onAccept(v!!)
            } else {
                callback.onReject(v!!)
            }
        }
    }
}